### PR TITLE
BadDet Remainder Attacks Implementation

### LINKED ISSUE #2038: Implementation of BadDet Poisoning Attack on Object Detectors
**Is your feature request related to a problem? Please describe.**
BadDet is a backdoor poisoning attack on object detection models such as Faster R-CNN and YOLO. This attack is a generalization of BadNet (the dirty label backdoor attack) and has shown to be effective in inserting backdoor triggers to degrade models. This attack has four variations:

1. Object Generation Attack (OGA)
2. Regional Misclassification Attack (RMA)
3. Global Misclassification Attack (GMA)
4. Object Disappearance Attack (ODA)

Ideally, all four attacks will be implemented eventually, but the first will begin with the simple Regional Misclassification Attack (RMA) which changes the label of a detection.

Paper link: <https://arxiv.org/abs/2205.14497>

**Describe the solution you'd like**
Under the `adversarial-robustness-toolbox/attacks/poisoning` directory, a subdirectory `bad_det` will be created. In this directory, an abstract class `BadDet` will extend `art.attacks.PoisoningAttackBlackBox` to serve as the base for the four attack variants. Additional classes (e.g., `BadDetOGA`, `BadDetRMA`, etc.) will be created in the `adversarial-robustness-toolbox/attacks/poisoning/bad_det` directory which extend the `BadDet` abstract class and implement the respective attack.

**Describe alternatives you've considered**
One idea is to create one class `art.attacks.poisoning.BadDet` which contains all four variants of the attack. However, this will make the class too complex as each variant is a unique attack and will require too much optional parameter parsing. It is simpler to keep each variant separate as their own class.

**Additional context**
It might be worth breaking this into multiple issues, one for each attack variant. This is mainly due to the fact that each variant of BadDet is essentially its own unique attack.

### PR DESCRIPTION
**Pull request by f4str**
# Description

Implementation of the remainder of the BadDet Attacks:
* BadDet Global Misclassification Attack (GMA)
* BadDet Object Generation Attack (OGA)
* BadDet Object Disappearance Attack (ODA)

Fixes a bug with the BadDet Regional Misclassification Attack (RMA) where it incorrectly requires the `scores` fields with an improvement on the implementation to allow for the attack to work regardless of the channel order.

Fixes another bug with the predict functionality in the `PyTorchYolo` estimator not using the `channels_first` parameter properly.

Changed the attack-specific notebook `notebooks/poisoning_attack_bad_det_rma.ipynb` to a general notebook `notebooks/poisoning_attack_bad_det.ipynb` which demonstrates how to use all four types of attacks.

Paper link: <https://arxiv.org/abs/2205.14497>

Fixes #2038 

## Type of change

Please check all relevant options.

- [x] Improvement (non-breaking)
- [x] Bug fix (non-breaking)
- [x] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Testing

Please describe the tests that you ran to verify your changes. Consider listing any relevant details of your test configuration.

- [x] Tests for BadDet Regional Misclassification Attack (RMA) which are framework agnostic
- [x] Tests for BadDet Global Misclassification Attack (GMA) which are framework agnostic
- [x] Tests for the BadDet Object Generation Attack (OGA) which are framework agnostic
- [x] Tests for BadDet Object Disappearance Attack (ODA) which are framework agnostic

**Test Configuration**:
- OS
- Python version
- ART version or commit number
- TensorFlow / Keras / PyTorch / MXNet version

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by beat-buesser — CHANGES_REQUESTED** [review]

File: art/attacks/poisoning/bad_det/bad_det_rma.py [hunk 1]
```diff
@@ -93,28 +93,28 @@ def poison(  # pylint: disable=W0221
         :param y: True labels of type `List[Dict[np.ndarray]]`, one dictionary per input image.
                   The keys and values of the dictionary are:
                   - boxes [N, 4]: the boxes in [x1, y1, x2, y2] format, with 0 <= x1 < x2 <= W and 0 <= y1 < y2 <= H.
-                  - labels [N]: the labels for each image
-                  - scores [N]: the scores or each prediction.
```
[the remark this answers is not in the document]
Comment (f4str) [reply art/attacks/poisoning/bad_det/bad_det_rma.py:97]: The `scores` field is not required for this attack since only `labels` and `boxes` are used. But for consistency, this can be re-added to the docstring.

File: art/attacks/poisoning/bad_det/bad_det_gma.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+    attack_params = PoisoningAttackObjectDetector.attack_params + [
+        "backdoor",
+        "source_class",
+        "target_class",
+        "percent_poison",
+        "channels_first",
+        "verbose",
+    ]
+    _estimator_requirements = ()
```
[the remark this answers is not in the document]
Comment (beat-buesser) [reply art/attacks/poisoning/bad_det/bad_det_gma.py:51]:
```suggestion
    _estimator_requirements = (BaseEstimator, ObjectDetectorMixin)
```

File: art/attacks/poisoning/bad_det/bad_det_oda.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+    attack_params = PoisoningAttackObjectDetector.attack_params + [
+        "backdoor",
+        "source_class",
+        "target_class",
+        "percent_poison",
+        "channels_first",
+        "verbose",
+    ]
+    _estimator_requirements = ()
```
[the remark this answers is not in the document]
Comment (beat-buesser) [reply art/attacks/poisoning/bad_det/bad_det_oda.py:51]:
```suggestion
    _estimator_requirements = (BaseEstimator, ObjectDetectorMixin)
```

File: art/attacks/poisoning/bad_det/bad_det_oga.py [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    attack_params = PoisoningAttackObjectDetector.attack_params + [
+        "backdoor",
+        "bbox_height",
+        "bbox_width",
+        "class_target",
+        "percent_poison",
+        "channels_first",
+        "verbose",
+    ]
+    _estimator_requirements = ()
```
Comment (beat-buesser) [inline art/attacks/poisoning/bad_det/bad_det_oga.py:53]:
```suggestion
    _estimator_requirements = (BaseEstimator, ObjectDetectorMixin)
```

Comment (beat-buesser) [reply art/attacks/poisoning/bad_det/bad_det_oga.py:53]: We also need to make the same updates to RMA.

**Review by beat-buesser — APPROVED** [review]
